skops/hub_utils/_hf_hub.py

```diff
@@ -327,6 +328,65 @@ def init(
         raise
 
 
+def add_files(files: Sequence[str | Path], dst: str | Path) -> None:
+    """Add files to initialized repo.
+
+    After having called :func:`.hub_utils.init`, use this function to add
+    arbitrary files to be uploaded in addition to the model and model card.
+
+    In particular, it can be useful to upload the script itself that produces
+    those artifacts by calling ``hub_utils.add_files([__file__], dst=...)``.
+
+    Parameters
+    ----------
+    files : list of str or Path
+        The files to be added. If a file by that name already exists at target
+        location, it will be skipped.
+
+    dst : str or Path
+        Path to the initialized repo, same as used during
+        :func:`.hub_utils.init`.
+
+    Raises
+    ------
+    FileNotFoundError
+        When the target folder or the files to be added are not found.
+
+    TypeError
+        When the files are not passed correctly.
+
+    """
+    dst = Path(dst)
+    # check dst exists
+    if not dst.exists():
+        msg = f"Could not find '{dst}', did you run 'skops.hub_utils.init' first?"
+        raise FileNotFoundError(msg)
+
+    # validate input types
+    if isinstance(files, (str, Path)):
+        msg = (
+            "First argument must be a sequence of str or Path, got "
+            f"{type(files)} instead."
+        )
+        raise TypeError(msg)
```

Comment:
even if we stick with the current signature, I think we should accept a simple string or Path here.